ts/components/MemberListItem.tsx

```diff
@@ -86,25 +116,204 @@ export const MemberListItem = (props: {
+      isFailure={
+        (groupPromotionFailed && !groupPromotionSending) ||
+        (groupInviteFailed && !groupInviteSending)
+      }
+    >
+      {statusText}
+    </StyledGroupStatusText>
+  );
+};
+
+const GroupStatusContainer = ({
+  displayGroupStatus,
+  groupPk,
+  pubkey,
+}: Pick<MemberListItemProps, 'displayGroupStatus' | 'pubkey' | 'groupPk'>) => {
+  if (
+    displayGroupStatus &&
+    groupPk &&
+    PubKey.is03Pubkey(groupPk) &&
+    PubKey.is05Pubkey(pubkey) &&
+    !UserUtils.isUsFromCache(pubkey)
+  ) {
+    return <GroupStatusText groupPk={groupPk} pubkey={pubkey} />;
+  }
+  return null;
+};
 
+const ResendButton = ({ groupPk, pubkey }: { pubkey: PubkeyType; groupPk: GroupPubkeyType }) => {
+  const acceptedInvite = useMemberHasAcceptedInvite(pubkey, groupPk);
+  const nominatedAdmin = useMemberIsNominatedAdmin(pubkey, groupPk);
+
+  // as soon as the `admin` flag is set in the group for that member, we should be able to resend a promote as we cannot remove an admin.
+  const canResendPromotion = hasClosedGroupV2QAButtons() && nominatedAdmin;
+
+  // we can always remove/and readd a non-admin member. So we consider that a member who accepted the invite cannot be resent an invite.
+  const canResendInvite = !acceptedInvite;
+
+  const shouldShowResendButton = canResendInvite || canResendPromotion;
+
+  if (!shouldShowResendButton) {
+    return null;
+  }
+  return (
+    <SessionButton
+      dataTestId={'resend-invite-button'}
+      buttonShape={SessionButtonShape.Square}
+      buttonType={SessionButtonType.Solid}
+      text={window.i18n('resend')}
+      onClick={async () => {
+        const group = await UserGroupsWrapperActions.getGroup(groupPk);
+        const member = await MetaGroupWrapperActions.memberGet(groupPk, pubkey);
+        if (!group || !group.secretKey || isEmpty(group.secretKey) || !member) {
+          window.log.warn('tried to resend invite but we do not have correct details');
+          return;
+        }
+        // if we tried to invite that member as admin right away, let's retry it as such.
+        const inviteAsAdmin = member.nominatedAdmin;
+        await GroupInvite.addJob({
+          groupPk,
+          member: pubkey,
+          inviteAsAdmin,
+          forceUnrevoke: true,
+        });
+      }}
+    />
+  );
+};
+
+const PromoteButton = ({ groupPk, pubkey }: { pubkey: PubkeyType; groupPk: GroupPubkeyType }) => {
+  const memberAcceptedInvite = useMemberHasAcceptedInvite(pubkey, groupPk);
+  const memberIsNominatedAdmin = useMemberIsNominatedAdmin(pubkey, groupPk);
+  const memberIsPendingRemoval = useMemberPendingRemoval(pubkey, groupPk);
+  // When invite-as-admin was used to invite that member, the resend button is available to resend the promote message.
+  // We want to show that button only to promote a normal member who accepted a normal invite but wasn't promoted yet.
+  // ^ this is only the case for testing. The UI will be different once we release the promotion process
```

Comment:
Maybe a TODO comment here?

Reply:
that's all hidden behind hasClosedGroupV2QAButtons that I'll remove before we built it properly.